### PR TITLE
Add schema translation from ShoppingCartItem to ProductContext 

### PR DESCRIPTION
**Pull request by benjaminkalk**
## Description

ProductContext for all items added or increased in quantity is not available at time of event, and therefore we need to do our best to populate the expected snowplow schema with the data we have.

## Related Issue

https://jira.corp.adobe.com/browse/DINT-531

## Motivation and Context

Need to update snowplow collector based on add to cart logic changes

## How Has This Been Tested?

Tested locally using Snowplow chrome plugin, and verified as good events in QA kibana

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] My change requires a change to the documentation.
-   [ ] I have updated the documentation accordingly.
-   [x] I have read the **CONTRIBUTING** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

### CODE REVIEW

File: packages/commerce-events-collectors/src/utils/product.ts [hunk 1]
```diff
@@ -23,4 +23,33 @@ const createPricing = (ctx: Product): ProductPricing | undefined => {
     return pricing;
 };
 
-export { createPricing };
+const createProductFromCartItem = (ctx: ShoppingCartItem): ProductContext => {
+    const pricing : ProductPricing = {
+        regularPrice: ctx.prices.price.value,
+        currencyCode: ctx.prices.price.currency || null,
+    };
+    let productId : any = ctx.product.productId;
```
Comment (benjaminkalk) [inline packages/commerce-events-collectors/src/utils/product.ts:31]: There was some weird issue where productId would be a string, and so needed to parse it, but due to types I have to explicitly set it to any before being able to parse it.

Comment (aniham) [reply packages/commerce-events-collectors/src/utils/product.ts:31]: yikes

i _think_ you might be able to do `let productId : string = ctx.product.productId as string` but... what do i know... ts continues to surprise me

File: packages/storefront-events-collector/src/handlers/product/addToCart.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ const handler = (event: Event): void => {
         if(item.product.sku === productContext.sku){
             productCtx = createProductCtx(productContext);
         } else {
-            productCtx = createProductCtx(item.product)
+            productCtx = createProductFromCartItem(item)
```
Comment (aniham) [inline packages/storefront-events-collector/src/handlers/product/addToCart.ts:13]: this is missing a semi-colon which in itself is not the end of the world but i don't understand why the linter didn't run here pre-commit... thoughts?

**Review by aniham — APPROVED** [review]

File: packages/storefront-events-collector/src/handlers/product/addToCart.ts (hunk 2 above)
Comment (benjaminkalk) [reply packages/storefront-events-collector/src/handlers/product/addToCart.ts:13]: I don't know if linter is set on this project.

**Pull request merged**